File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/budget/dto/BudgetResponse.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.budget.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
public class BudgetResponse {

    private String patientDni;
    private List<Budget> budgets;

    public BudgetResponse() {
        budgets = new ArrayList<>();
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Budget {
        private Integer id;
        private Date date;
        private Status status;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Status {
        private Integer id;
        private String description;
    }

}

File: front/dentalium/src/app/model/api/BudgetResponse.ts
export class BudgetResponse {
  patientDni: string;
  budgets: Budget[];

  constructor() {
    this.patientDni = "";
    this.budgets = [];
  }
}

export class Budget {
  id: number;
  date: Date;
  status: Status;

  constructor(id: number, date: Date, status: Status) {
    this.id = id;
    this.date = date;
    this.status = status;
  }
}

export class Status {
  id: number;
  description: string;

  constructor(id: number, description: string) {
    this.id = id;
    this.description = description;
  }
}
